sponge/src/main/java/com/vexsoftware/votifier/sponge/config/ConfigLoader.java

```diff
@@ -0,0 +1,45 @@
+package com.vexsoftware.votifier.sponge.config;
+
+import com.google.common.reflect.TypeToken;
+import com.vexsoftware.votifier.sponge.VotifierPlugin;
+import ninja.leaping.configurate.ConfigurationNode;
+import ninja.leaping.configurate.ConfigurationOptions;
+import ninja.leaping.configurate.commented.CommentedConfigurationNode;
+import ninja.leaping.configurate.loader.ConfigurationLoader;
+import ninja.leaping.configurate.yaml.YAMLConfigurationLoader;
+
+import java.io.File;
+
+public class ConfigLoader {
+
+    private final VotifierPlugin plugin;
+    private SpongeConfig spongeConfig;
+
+    public ConfigLoader(VotifierPlugin pl) {
+        this.plugin = pl;
+        if (!plugin.getConfigDir().exists()) {
```

Comment:
This should live down with loadConfig - since it mutates state, it would be useful to be in the same spot as the rest of our filesystem state mutating code.


On a seperate more practical note, imagine a situation where a user deleted the entire configuration directory and reloaded - we should create the directory for the configuration again.

Reply:
I'll see if I can move the constructor down into load config. Another note, this plugin doesn't reload, if you would like me to add it I can most certainly do that.

Reply:
If you want to do that in another PR that would be great. Once this change is made I’ll merge this in.